cmd/sqlcmd/sqlcmd.go

```diff
@@ -117,25 +118,140 @@ func (a SQLCmdArguments) authenticationMethod(hasPassword bool) string {
 }
 
 func Execute(version string) {
-	ctx := kong.Parse(&args, kong.NoDefaultHelp())
-	if args.Version {
-		ctx.Printf("%v", version)
-		os.Exit(0)
+	// Cobra Tryout #2 : This has version and help flags. But they are also getting stucked inside run function without error
+	rootCmd := &cobra.Command{
+		Use:   "testCommand",
+		Short: "A brief description of your command",
+		Run: func(cmd *cobra.Command, argss []string) {
+			vars := sqlcmd.InitializeVariables(!args.DisableCmdAndWarn)
+			setVars(vars, &args)
+
+			if args.Version {
+				fmt.Printf("%v\n", version)
+				os.Exit(0)
+			}
+
+			if args.Help {
+				fmt.Print(cmd.UsageString())
+				os.Exit(0)
+			}
+
+			exitCode, err := run(vars, &args)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			fmt.Printf("after exit exitCode: %d\n", exitCode)
+			//os.Exit(exitCode)
+			fmt.Printf("after execute Function")
+		},
 	}
-	if args.Help {
-		_ = ctx.PrintUsage(false)
-		os.Exit(0)
+	setFlags(rootCmd, &args)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
 	}
-	vars := sqlcmd.InitializeVariables(!args.DisableCmdAndWarn)
-	setVars(vars, &args)
+}
+
+func setFlags(rootCmd *cobra.Command, args *SQLCmdArguments) {
+	var inputfiles []string
+	rootCmd.Flags().StringArrayVarP(&args.InputFile, "i", "i", inputfiles, "input file")
+	rootCmd.PersistentFlags().BoolVarP(&args.Version, "Version", "", false, "Print version information and exit")
```

Comment:
>PersistentFlags [](http://example.com/codeflow?start=9&length=15)

since we don't have any child commands, can't all these just be Flags() instead of PersistentFlags() ?